lib/fluent/plugin/out_forward.rb

```diff
@@ -401,12 +394,15 @@ def on_timer
       }
     end
 
-    def on_heartbeat(sockaddr, msg)
-      if node = @nodes.find {|n| n.sockaddr == sockaddr }
+    def on_udp_heatbeat_response_recv(data, sock)
+      sockaddr = Socket.pack_sockaddr_in(sock.remote_port, sock.remote_host)
+      if node = @nodes.find { |n| n.sockaddr == sockaddr }
         # log.trace "heartbeat arrived", name: node.name, host: node.host, port: node.port
         if node.heartbeat
           @load_balancer.rebuild_weight_array(@nodes)
         end
+      else
+        log.warn("Unknown heartbeat response received host: #{soc.remote_host}, port: #{sock.remote_port}")
```

Comment:
`received from #{sock.remote_host}:#{sock.remote_port}` is more readable for me.

Reply:
https://github.com/fluent/fluentd/pull/2503/commits/efa756543e4e982145b4c17bcdaec007bf2520ea